### PR TITLE
Removed jaggery.js component from admin portal and replaced it with jsp. Migrated the admin portal as a webapp.

### PR DESCRIPTION
**Pull request by DinithHerath**

### CODE REVIEW

**Review by tharikaGitHub — APPROVED** [review]

File: portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/*
+
```
Comment (rmsamitha) [inline portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java:2]: Remove additional empty lines

File: portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+ *  Copyright (c) 2017, WSO2 LLC (http://www.wso2.org) All Rights Reserved.
+ *
```
Comment (rmsamitha) [inline portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java:5]: Incorrect license header

File: portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+
+public class Util {
+
+    /**
+     * Read a json file from the directory and output as a Map object.
+     *
+     * @param path    required parameter
+     * @param context required parameter
+     * @return Map<String, Object>
```
Comment (rmsamitha) [inline portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java:42]: Return type is already obvious from the code. Provide some description. (or remove the line completely if there is no useful description)

File: portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+     * @throws FileNotFoundException
```
Comment (rmsamitha) [inline portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java:43]: Exception is already obvious from the code. Provide some description. (or remove the line completely if there is no useful description)

File: portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        }
+        return nestedJson.get(pathStrings[pathStrings.length - 1]);
+    }
+    
+    /**
+     * Get the loopback (localhost) origin (scheme + hostname + port), This origin is used for making
+     * internal(within the web app node), API calls.For example DCR call, Token generation, User Info, Token Introspect,
+ Revoke etc.
+     * @param host
+     * @return String
```
Comment (rmsamitha) [inline portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java:81]: Return type is already obvious from the code. Provide some description. (or remove the line completely if there is no useful description). Fix this in other places too in your code, considering other params too

File: portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+     *     context: '/devportal'
+     * </pre>
+     *
+     * <p> ( case 2 ) - appContext is <b>'/apim/devportal'</b>
+     * <pre>
+     *     context: '/devportal'
+     *     proxy_context_path: '/apim',
+     * </pre>
+     *
+     * @param proxyContext required parameter
```
Comment (rmsamitha) [inline portals/admin/src/main/java/org/wso2/carbon/apimgt/ui/admin/Util.java:110]: Always params are required in java

File: portals/admin/src/main/webapp/WEB-INF/web.xml [hunk 7]
```diff
@@ -0,0 +1,198 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--  
+    ~ Copyright (c) 2017, WSO2 LLC (http://www.wso2.org) All Rights Reserved.
```
Comment (rmsamitha) [inline portals/admin/src/main/webapp/WEB-INF/web.xml:3]: Incorrect license header

File: portals/admin/src/main/webapp/services/constants.jsp [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<%--
+  ~ Copyright (c) 2017, WSO2 LLC (http://www.wso2.org) All Rights Reserved.
+  ~
```
Comment (rmsamitha) [inline portals/admin/src/main/webapp/services/constants.jsp:3]: Incorrect license header. Fix at other places too

File: pom.xml [hunk 9]
```diff
@@ -146,5 +188,12 @@
...
+        <!-- Dependancy versions -->
+        <javax.servlet-api.version>3.1.0</javax.servlet-api.version>
+        <javax.servlet.jsp-api.version>2.3.1</javax.servlet.jsp-api.version>
+        <javaee-web-api.version>7.0</javaee-web-api.version>
+        <com.google.code.gson.version>2.9.0</com.google.code.gson.version>
+        <carbon.apimgt.imp.pkg.version.range>9.22.32</carbon.apimgt.imp.pkg.version.range>
+        <commons-logging.version>1.2</commons-logging.version>
     </properties>
-</project>
+</project>
```
Comment (SanojPunchihewa) [inline pom.xml:199]: Missing newline

**Pull request merged**